pkg/kn/commands/service/export_test.go

```diff
@@ -35,14 +35,15 @@ type expectedServiceOption func(*servingv1.Service)
 type expectedRevisionOption func(*servingv1.Revision)
 type expectedServiceListOption func(*servingv1.ServiceList)
 type expectedRevisionListOption func(*servingv1.RevisionList)
+type expectedKNExportOption func(*clientv1alpha1.Export)
 type podSpecOption func(*v1.PodSpec)
 
 type testCase struct {
-	name                 string
-	latestSvc            *servingv1.Service
-	expectedSvcList      *servingv1.ServiceList
-	revisionList         *servingv1.RevisionList
-	expectedRevisionList *servingv1.RevisionList
+	name             string
+	latestSvc        *servingv1.Service
+	expectedSvcList  *servingv1.ServiceList
+	revisionList     *servingv1.RevisionList
+	expectedKNExport *clientv1alpha1.Export
```

Comment:
Would be nice to have the expected YAML output in files or in vars and break down the top test into `export` call and verify with YAML. That makes it easy to read and debug since the rest is supporting functions. WDYT?

Reply:
for sure having the files would make it easier to read. Was just following how it was in the other knative repo's. Like this [one](https://github.com/knative/serving/blob/master/pkg/reconciler/serverlessservice/serverlessservice_test.go) .

Reply:
I'm all for reusing and aligning with serving/eventing way of testing code generation, even when I see the potential benefits of having human readable YAML files. The advantage though of having that in code is that it wlll immediately will break when the API changes, not so clear to debug when using it with YAML file.

So I'm fine with this approach here.